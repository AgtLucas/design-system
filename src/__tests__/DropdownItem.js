import React from 'react'
import renderer from 'react-test-renderer'
import { DropdownItem, ThemeProvider } from '..'

describe('DropdownItem', () => {
  test('renders with itemIdenx prop', () => {
    const json = renderer
      .create(
        <ThemeProvider>
          <DropdownItem itemIndex={1} isSelected>
            Test 1
          </DropdownItem>
        </ThemeProvider>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with isSelected prop', () => {
    const json = renderer
      .create(
        <ThemeProvider>
          <DropdownItem isSelected>Test 2</DropdownItem>
        </ThemeProvider>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders without any prop', () => {
    const json = renderer
      .create(
        <ThemeProvider>
          <DropdownItem>Test 3</DropdownItem>
        </ThemeProvider>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })
})
