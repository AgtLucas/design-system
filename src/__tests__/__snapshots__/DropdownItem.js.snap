// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownItem renders with isSelected prop 1`] = `
.c1 {
  list-style: none;
  background-color: #007aff;
  color: #fff;
}

.c1:hover {
  background-color: #007aff;
  color: #fff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c0 * {
  box-sizing: border-box;
}

<div
  className="c0"
>
  <li
    className="c1"
    id={null}
    role="menuitem"
    tabIndex={-1}
  >
    Test 2
  </li>
</div>
`;

exports[`DropdownItem renders with itemIdenx prop 1`] = `
.c1 {
  list-style: none;
  background-color: #007aff;
  color: #fff;
}

.c1:hover {
  background-color: #007aff;
  color: #fff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c0 * {
  box-sizing: border-box;
}

<div
  className="c0"
>
  <li
    className="c1"
    id="mi1"
    role="menuitem"
    tabIndex={-1}
  >
    Test 1
  </li>
</div>
`;

exports[`DropdownItem renders without any prop 1`] = `
.c1 {
  list-style: none;
}

.c1:hover {
  background-color: #007aff;
  color: #fff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c0 * {
  box-sizing: border-box;
}

<div
  className="c0"
>
  <li
    className="c1"
    id={null}
    role="menuitem"
    tabIndex={-1}
  >
    Test 3
  </li>
</div>
`;
